test/classification/test_snorkel_classifier.py

```diff
@@ -83,6 +83,44 @@ def test_predict(self):
             np.array([0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0]),
         )
 
+    def test_predict_rerouted_label(self):
+        """Label name task:other_label should reroute to task if not in task_flow."""
+        task = create_task("task", module_suffixes=["A", "A"])
+        model = SnorkelClassifier([task])
+
+        # Create dataloader with two labelsets
+        X = torch.FloatTensor([[i, i] for i in range(NUM_EXAMPLES)])
+        Y = torch.ones(NUM_EXAMPLES, 1).long()
+        dataset = DictDataset(
+            name="dataset",
+            split="train",
+            X_dict={"data": X},
+            Y_dict={"task:other_label": Y},
+        )
+        dataloader = DictDataLoader(dataset, batch_size=BATCH_SIZE)
+
+        results = model.predict(dataloader)
+        self.assertEqual(sorted(list(results.keys())), ["golds", "probs"])
+        self.assertIn("task:other_label", results["golds"])
+        self.assertIn("task:other_label", results["probs"])
+        self.assertNotIn("task", results["golds"])
+        self.assertNotIn("task", results["probs"])
+        np.testing.assert_array_equal(
+            results["probs"]["task:other_label"], np.ones((NUM_EXAMPLES, 2)) * 0.5
+        )
+
+        # Now, test malformed version
+        Y = torch.ones(NUM_EXAMPLES, 1).long()
+        dataset = DictDataset(
+            name="dataset",
+            split="train",
+            X_dict={"data": X},
+            Y_dict={"task:other_label:bad!": Y},
```

Comment:
I'm lazy...do we already have a test for a bad name without `:`?

Reply:
without the `:` is treated as a normal labelset, so it's covered in other test cases